### DISCUSSION TITLE
Reindex operation?

### DISCUSSION THREAD

**Discussion by KMCGit** [FirelyTeam/spark#514] [Q&A] [answered]
Good Day, I wanted to quickly ask: If I wanted to add a Search Parameter, what would be the proper way to reindex with the Spark server? 

All help and guidance is appreciated. Thanks and have a great day.

**Comment by kennethmyhra** [comment, marked as answer]
Hey,

If you are using Spark.Web you can use the Login button at the top, then click the Admin button that appears and you will get access to the various administrative function in the Admin GUI. You can find the default username and password here: https://github.com/FirelyTeam/spark/blob/02125beb85f5455450a391cbc5b57aeb7b06c48a/src/Spark.Web/appsettings.json#L11-L14

If you are using the Spark-Legacy project you can go to this URL: `/Maintenance/RebuildIndex`

**Reply by KMCGit** [reply]
Thanks for the quick answer. I was able to access the Admin panel and "Rebuild index". Before doing this, I added a custom Search Parameter to /SearchParameters, but it was not operational afterwards.

POST {{FHIR_URL}}/SearchParameter

{
  "resourceType" : "SearchParameter",
  "id" : "us-core-race",
  "url" : "http://hl7.org/fhir/us/core/SearchParameter/us-core-race",
  "version" : "3.1.1",
  "name" : "USCoreRace",
  "status" : "active",
  "date" : "2019-05-21",
  "publisher" : "US Realm Steering Committee",
  "contact" : [
    {
      "telecom" : [
        {
          "system" : "other",
          "value" : "http://www.healthit.gov/"
        }
      ]
    }
  ],
  "description" : "Returns patients with a race extension matching the specified code.",
  "jurisdiction" : [
    {
      "coding" : [
        {
          "system" : "urn:iso:std:iso:3166",
          "code" : "US",
          "display" : "United States of America"
        }
      ]
    }
  ],
  "code" : "race",
  "base" : [
    "Patient"
  ],
  "type" : "token",
  "expression" : "Patient.extension.where(url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race').extension.value.code"
}

What would be the best/proper way to add a Custom Search parameter? 
(Thanks for your assistance in this matter, it is much appreciated.)

**Reply by kennethmyhra** [reply]
Answered here: https://github.com/FirelyTeam/spark/discussions/516#discussioncomment-3903218
